{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### This notebook is to create the preparation information and sample information files for Qiita, as well as the mapping file for Qiime."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## create preparation_info document"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "info = pd.read_csv('../data/Demultiplex_Sheet.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#SampleID</th>\n",
       "      <th>BarcodeSequence</th>\n",
       "      <th>LinkerPrimerSequence</th>\n",
       "      <th>ReversePrimer</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "      <td>TCTGGTGACATT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1129</th>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "      <td>TCTGGTGACATT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "      <td>CAAGCATGCCTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475</th>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "      <td>CAAGCATGCCTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "      <td>CTATTTGCGACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             #SampleID BarcodeSequence  LinkerPrimerSequence  \\\n",
       "60    Orwoll.BI0023.BI    TCTGGTGACATT  GGACTACHVGGGTWTCTAAT   \n",
       "1129  Orwoll.BI0023.BI    TCTGGTGACATT  GGACTACHVGGGTWTCTAAT   \n",
       "649   Orwoll.BI0056.BI    CAAGCATGCCTA  GGACTACHVGGGTWTCTAAT   \n",
       "475   Orwoll.BI0056.BI    CAAGCATGCCTA  GGACTACHVGGGTWTCTAAT   \n",
       "75    Orwoll.BI0131.BI    CTATTTGCGACA  GGACTACHVGGGTWTCTAAT   \n",
       "\n",
       "            ReversePrimer       Description  \n",
       "60    GTGCCAGCMGCCGCGGTAA  Orwoll.BI0023.BI  \n",
       "1129  GTGCCAGCMGCCGCGGTAA  Orwoll.BI0023.BI  \n",
       "649   GTGCCAGCMGCCGCGGTAA  Orwoll.BI0056.BI  \n",
       "475   GTGCCAGCMGCCGCGGTAA  Orwoll.BI0056.BI  \n",
       "75    GTGCCAGCMGCCGCGGTAA  Orwoll.BI0131.BI  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info = info.sort_values(['#SampleID'])\n",
    "info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "info = info.drop_duplicates(subset=['#SampleID', 'ReversePrimer'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#SampleID</th>\n",
       "      <th>BarcodeSequence</th>\n",
       "      <th>LinkerPrimerSequence</th>\n",
       "      <th>ReversePrimer</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "      <td>TCTGGTGACATT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "      <td>CAAGCATGCCTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "      <td>CTATTTGCGACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>932</th>\n",
       "      <td>Orwoll.BI0153.BI</td>\n",
       "      <td>ATCGGCGTTACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0153.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>833</th>\n",
       "      <td>Orwoll.BI0215.BI</td>\n",
       "      <td>CCTCTCGTGATC</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>GTGCCAGCMGCCGCGGTAA</td>\n",
       "      <td>Orwoll.BI0215.BI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            #SampleID BarcodeSequence  LinkerPrimerSequence  \\\n",
       "60   Orwoll.BI0023.BI    TCTGGTGACATT  GGACTACHVGGGTWTCTAAT   \n",
       "649  Orwoll.BI0056.BI    CAAGCATGCCTA  GGACTACHVGGGTWTCTAAT   \n",
       "75   Orwoll.BI0131.BI    CTATTTGCGACA  GGACTACHVGGGTWTCTAAT   \n",
       "932  Orwoll.BI0153.BI    ATCGGCGTTACA  GGACTACHVGGGTWTCTAAT   \n",
       "833  Orwoll.BI0215.BI    CCTCTCGTGATC  GGACTACHVGGGTWTCTAAT   \n",
       "\n",
       "           ReversePrimer       Description  \n",
       "60   GTGCCAGCMGCCGCGGTAA  Orwoll.BI0023.BI  \n",
       "649  GTGCCAGCMGCCGCGGTAA  Orwoll.BI0056.BI  \n",
       "75   GTGCCAGCMGCCGCGGTAA  Orwoll.BI0131.BI  \n",
       "932  GTGCCAGCMGCCGCGGTAA  Orwoll.BI0153.BI  \n",
       "833  GTGCCAGCMGCCGCGGTAA  Orwoll.BI0215.BI  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "600"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "n = info.shape[0]\n",
    "data = {'#SampleID': info.loc[:,'#SampleID'].str[7:13].values, # .values takes only value not index\n",
    "       'Barcode': info.loc[:,'BarcodeSequence'].values,\n",
    "       'LinkerPrimerSequence': info.loc[:, 'LinkerPrimerSequence'].values,\n",
    "       'Description': info.loc[:,'Description'].values,\n",
    "       'Experiment_Design_Description': np.repeat(\n",
    "           '16S stool samples sequenced for MrOS Vitamin D study', n),\n",
    "       'Library_Construction_Protocol': np.repeat('16S rRNA v4', n),\n",
    "       'Linker': np.repeat('GT', n),\n",
    "       'Platform': np.repeat('Illumina', n),\n",
    "       'Center_Name': info.loc[:,'#SampleID'].str[14:16].values,\n",
    "       'Center_Project': np.repeat('MrOS', n),\n",
    "       'Instrument_Model': np.repeat('Illumina MiSeq', n)}\n",
    "prep_info = pd.DataFrame(data, columns=['#SampleID', 'Barcode', 'LinkerPrimerSequence', \n",
    "                                        'Description', \n",
    "                                        'Experiment_Design_Description',\n",
    "                                        'Library_Construction_Protocol', 'Linker', 'Platform',\n",
    "                                        'Center_Name', 'Center_Project', 'Instrument_Model'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(600, 11)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prep_info.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#SampleID</th>\n",
       "      <th>Barcode</th>\n",
       "      <th>LinkerPrimerSequence</th>\n",
       "      <th>Description</th>\n",
       "      <th>Experiment_Design_Description</th>\n",
       "      <th>Library_Construction_Protocol</th>\n",
       "      <th>Linker</th>\n",
       "      <th>Platform</th>\n",
       "      <th>Center_Name</th>\n",
       "      <th>Center_Project</th>\n",
       "      <th>Instrument_Model</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BI0023</td>\n",
       "      <td>TCTGGTGACATT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BI0056</td>\n",
       "      <td>CAAGCATGCCTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BI0131</td>\n",
       "      <td>CTATTTGCGACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BI0153</td>\n",
       "      <td>ATCGGCGTTACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0153.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BI0215</td>\n",
       "      <td>CCTCTCGTGATC</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0215.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BI0353</td>\n",
       "      <td>TGCCATCTGAAT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0353.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>BI0371</td>\n",
       "      <td>GAATAGAGCCAA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0371.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BI0372</td>\n",
       "      <td>ATGCCAACCAAC</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0372.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>BI0380</td>\n",
       "      <td>GAGTCGCGTTTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0380.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>BI0389</td>\n",
       "      <td>GCTATGGACCGA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0389.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #SampleID       Barcode  LinkerPrimerSequence       Description  \\\n",
       "0    BI0023  TCTGGTGACATT  GGACTACHVGGGTWTCTAAT  Orwoll.BI0023.BI   \n",
       "1    BI0056  CAAGCATGCCTA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0056.BI   \n",
       "2    BI0131  CTATTTGCGACA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0131.BI   \n",
       "3    BI0153  ATCGGCGTTACA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0153.BI   \n",
       "4    BI0215  CCTCTCGTGATC  GGACTACHVGGGTWTCTAAT  Orwoll.BI0215.BI   \n",
       "5    BI0353  TGCCATCTGAAT  GGACTACHVGGGTWTCTAAT  Orwoll.BI0353.BI   \n",
       "6    BI0371  GAATAGAGCCAA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0371.BI   \n",
       "7    BI0372  ATGCCAACCAAC  GGACTACHVGGGTWTCTAAT  Orwoll.BI0372.BI   \n",
       "8    BI0380  GAGTCGCGTTTA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0380.BI   \n",
       "9    BI0389  GCTATGGACCGA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0389.BI   \n",
       "\n",
       "                       Experiment_Design_Description  \\\n",
       "0  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "1  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "2  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "3  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "4  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "5  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "6  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "7  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "8  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "9  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "\n",
       "  Library_Construction_Protocol Linker  Platform Center_Name Center_Project  \\\n",
       "0                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "1                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "2                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "3                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "4                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "5                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "6                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "7                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "8                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "9                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "\n",
       "  Instrument_Model  \n",
       "0   Illumina MiSeq  \n",
       "1   Illumina MiSeq  \n",
       "2   Illumina MiSeq  \n",
       "3   Illumina MiSeq  \n",
       "4   Illumina MiSeq  \n",
       "5   Illumina MiSeq  \n",
       "6   Illumina MiSeq  \n",
       "7   Illumina MiSeq  \n",
       "8   Illumina MiSeq  \n",
       "9   Illumina MiSeq  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prep_info.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# fix mismatch sample name 'BI0778' should be 'BIO778' (Nora's email on08/07/2017)\n",
    "prep_info = prep_info.replace(to_replace='BIO778', value='BI0778')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create sample_info document"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# note AMPHIT_15SD was missing in Nov6th metadata, thus copied from file Variable_sleepV4updated.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "samples = pd.read_csv('../data/SleepMetadataV4_Nov6th2017.csv', sep=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(600, 38)"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1: <=6 HRS    153\n",
       "2: 7 HRS      144\n",
       "3:>=8 HRS     128\n",
       "Name: SLEEPHRS, dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# error in categories of SLEEPHRS (additional space in second '2')\n",
    "samples.SLEEPHRS.value_counts()\n",
    "\n",
    "# correct\n",
    "samples.loc[samples.SLEEPHRS == '2: 7 HRS ', 'SLEEPHRS'] = '2: 7 HRS'\n",
    "\n",
    "# check\n",
    "samples.SLEEPHRS.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "M:Not Applicable                   231\n",
       "4: 6-13 drinks per week            106\n",
       "3: 3-5 drinks per week              90\n",
       "1: Less than one drink per week     78\n",
       "2: 1-2drinks per week               66\n",
       "5: 14 or more drinks per week       27\n",
       "K:Main Var:DK/UN                     2\n",
       "Name: TUDRAMT, dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples.TUDRAMT.value_counts() # alcohol drinking variable is wrong"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# replace alcohol drinking variable\n",
    "tmp = pd.read_csv('../data/mapping_vitd_v4.txt', sep='\\t')\n",
    "tmp = tmp[['ID', 'TUDRAMT']]\n",
    "tmp.rename(columns={'TUDRAMT':'TUDRAMT_REVISED'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(599, 2)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0:None drinker                     230\n",
       "4: 6-13 drinks per week            106\n",
       "3: 3-5 drinks per week              90\n",
       "1: Less than one drink per week     78\n",
       "2: 1-2drinks per week               66\n",
       "5: 14 or more drinks per week       27\n",
       ".                                    2\n",
       "Name: TUDRAMT_REVISED, dtype: int64"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp.TUDRAMT_REVISED.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>TUDRAMT_REVISED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BI0023</td>\n",
       "      <td>1: Less than one drink per week</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BI0056</td>\n",
       "      <td>0:None drinker</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BI0131</td>\n",
       "      <td>0:None drinker</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       ID                  TUDRAMT_REVISED\n",
       "0  BI0023  1: Less than one drink per week\n",
       "1  BI0056                   0:None drinker\n",
       "2  BI0131                   0:None drinker"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>AMPHIT_15SD</th>\n",
       "      <th>SLSLPHRS</th>\n",
       "      <th>SLEEPHRS</th>\n",
       "      <th>AMAMPT</th>\n",
       "      <th>AMPHIT</th>\n",
       "      <th>AMFVT</th>\n",
       "      <th>MIDATA</th>\n",
       "      <th>GIERACE</th>\n",
       "      <th>V4DATE</th>\n",
       "      <th>...</th>\n",
       "      <th>PQPEFFCY</th>\n",
       "      <th>PQPEFFIC</th>\n",
       "      <th>PQPSQI</th>\n",
       "      <th>PQBADSLP</th>\n",
       "      <th>HWBMI</th>\n",
       "      <th>TURSMOKE</th>\n",
       "      <th>M1ADEPR</th>\n",
       "      <th>M1BENZO</th>\n",
       "      <th>AMAMPT_C1</th>\n",
       "      <th>AMFVT_C1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BI0023</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1: &lt;=6 HRS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1:WHITE</td>\n",
       "      <td>9/10/15</td>\n",
       "      <td>...</td>\n",
       "      <td>93.33</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>28.890119</td>\n",
       "      <td>M:Not Applicable</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BI0056</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>176.175855</td>\n",
       "      <td>11.959884</td>\n",
       "      <td>1049.233855</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1:WHITE</td>\n",
       "      <td>5/28/15</td>\n",
       "      <td>...</td>\n",
       "      <td>60</td>\n",
       "      <td>3:&lt;65% EFFICIENCY</td>\n",
       "      <td>9.0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>28.539796</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BI0131</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>271.742879</td>\n",
       "      <td>10.997060</td>\n",
       "      <td>2069.032006</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1:WHITE</td>\n",
       "      <td>4/14/15</td>\n",
       "      <td>...</td>\n",
       "      <td>90</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>25.014242</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       ID  AMPHIT_15SD  SLSLPHRS    SLEEPHRS      AMAMPT     AMPHIT  \\\n",
       "0  BI0023          NaN       6.0  1: <=6 HRS         NaN        NaN   \n",
       "1  BI0056          0.0       NaN         NaN  176.175855  11.959884   \n",
       "2  BI0131          0.0       NaN         NaN  271.742879  10.997060   \n",
       "\n",
       "         AMFVT  MIDATA  GIERACE   V4DATE   ...    PQPEFFCY           PQPEFFIC  \\\n",
       "0          NaN  1: Yes  1:WHITE  9/10/15   ...       93.33  0:>85% EFFICIENCY   \n",
       "1  1049.233855  1: Yes  1:WHITE  5/28/15   ...          60  3:<65% EFFICIENCY   \n",
       "2  2069.032006  1: Yes  1:WHITE  4/14/15   ...          90  0:>85% EFFICIENCY   \n",
       "\n",
       "  PQPSQI PQBADSLP      HWBMI          TURSMOKE M1ADEPR M1BENZO AMAMPT_C1  \\\n",
       "0    4.0    0: No  28.890119  M:Not Applicable   0: No   0: No       NaN   \n",
       "1    9.0   1: Yes  28.539796            1:PAST   0: No   0: No       1.0   \n",
       "2    2.0    0: No  25.014242            1:PAST   0: No   0: No       1.0   \n",
       "\n",
       "  AMFVT_C1  \n",
       "0      NaN  \n",
       "1      1.0  \n",
       "2      1.0  \n",
       "\n",
       "[3 rows x 38 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "samples = pd.merge(samples, tmp, left_on='ID', right_on='ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ID', 'AMPHIT_15SD', 'SLSLPHRS', 'SLEEPHRS', 'AMAMPT', 'AMPHIT',\n",
       "       'AMFVT', 'MIDATA', 'GIERACE', 'V4DATE', 'SITE', 'GIMSTAT', 'MHDIAB',\n",
       "       'MHRHEU1', 'MHOA', 'MHCHF', 'MHMI', 'MHDEPR', 'MHDEPRT', 'TUDRAMT',\n",
       "       'PQPSLMED', 'PQPSQUAL', 'HWWGT', 'V4AGE1', 'HWHGT', 'PASCORE', 'QLCOMP',\n",
       "       'PQPSLDUR', 'PQPEFFCY', 'PQPEFFIC', 'PQPSQI', 'PQBADSLP', 'HWBMI',\n",
       "       'TURSMOKE', 'M1ADEPR', 'M1BENZO', 'AMAMPT_C1', 'AMFVT_C1',\n",
       "       'TUDRAMT_REVISED'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(599, 39)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "599"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pd.unique(samples.ID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# reference: http://www.latlong.net/\n",
    "Latitude = []\n",
    "Longitude = []\n",
    "sites = samples.loc[:, 'SITE']\n",
    "for i in range(samples.shape[0]):\n",
    "    if sites[i] == 'Birmingham':\n",
    "        latitude = '33.520661'\n",
    "        longitude = '-86.80249'\n",
    "    elif sites[i] == 'San Diego':\n",
    "        latitude = '32.715738'\n",
    "        longitude = '-117.1611'\n",
    "    elif sites[i] == 'Pittsburgh':\n",
    "        latitude = '40.440625'\n",
    "        longitude = '-79.99589'\n",
    "    elif sites[i] == 'Palo Alto':\n",
    "        latitude = '37.441883'\n",
    "        longitude = '-122.143'\n",
    "    elif sites[i] == 'Portland':\n",
    "        latitude = '45.523062'\n",
    "        longitude = '-122.6765'    \n",
    "    elif sites[i] == 'Minneapolis':\n",
    "        latitude = '44.977753'\n",
    "        longitude = '-93.26501'  \n",
    "    Latitude.append(latitude)\n",
    "    Longitude.append(longitude)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1       Birmingham\n",
      "90     Minneapolis\n",
      "200      Palo Alto\n",
      "300     Pittsburgh\n",
      "400       Portland\n",
      "500      San Diego\n",
      "Name: SITE, dtype: object\n",
      "['33.520661' '44.977753' '37.441883' '40.440625' '45.523062' '32.715738']\n",
      "['-86.80249' '-93.26501' '-122.143' '-79.99589' '-122.6765' '-117.1611']\n"
     ]
    }
   ],
   "source": [
    "# simple check\n",
    "print(samples['SITE'][[1,90,200,300,400, 500]])\n",
    "print(np.array(Latitude)[[1,90,200,300,400, 500]])\n",
    "print(np.array(Longitude)[[1,90,200,300,400, 500]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "m = samples.shape[0]\n",
    "required = {'#SampleID': samples.loc[:,'ID'].values,\n",
    "        'Title': np.repeat('MrOS_VitaminD', m),\n",
    "        'Anonymized_Name': samples.loc[:,'ID'].values,\n",
    "        'Scientific_Name': np.repeat('human gut metagenome', m),\n",
    "        'Taxon_ID': np.repeat('Not applicable', m),\n",
    "        #'Description': np.repeat('Not applicable', m),\n",
    "        'Sample_Type': np.repeat('stool', m),\n",
    "        'Geo_Loc_Name': samples.loc[:, 'SITE'].values,\n",
    "        'Elevation': np.repeat('Not applicable', m),\n",
    "        'Env_Biome': np.repeat('urban biome', m),\n",
    "        'Env_Feature': np.repeat('human-associated habitat', m),\n",
    "        'Env_Material': np.repeat('feces', m),\n",
    "        'Env_Package': np.repeat('human-gut', m),\n",
    "        'Latitude': Latitude,\n",
    "        'Longitude': Longitude,\n",
    "        'Collection_Timestamp': np.repeat('Not applicable', m),\n",
    "        'DNA_Extracted': np.repeat('Not applicable', m),\n",
    "        'Physical_Specimen_Location': np.repeat('Not applicable', m),\n",
    "        'Physical_Specimen_Remaining': np.repeat('Not applicable', m),\n",
    "        'Age': samples.loc[:,'V4AGE1'].values,\n",
    "        'Age_Units': np.repeat('years', m),\n",
    "        'Host_Subject_ID': np.repeat('Not applicable', m),\n",
    "        'Host_Taxid': np.repeat('Not applicable', m),\n",
    "        'Host_Scientific_Name': np.repeat('Homo sapiens', m),\n",
    "        'Host_Common_Name': np.repeat('human', m),\n",
    "        'Life_Stage': np.repeat('adult', m),\n",
    "        'Sex': np.repeat('male', m),\n",
    "        'Height': samples.loc[:, 'HWHGT'].values,\n",
    "        'Height_Units': np.repeat('cm', m),\n",
    "        'Weight': samples.loc[:, 'HWWGT'].values,\n",
    "        'Weight_Units': np.repeat('kg', m),  \n",
    "        'BMI': samples.loc[:, 'HWBMI'].values,\n",
    "        'Body_Habitat': np.repeat('UBERON:feces', m),\n",
    "        'Body_Site': np.repeat('UBERON:feces', m),\n",
    "        'Body_Product': np.repeat('UBERON:feces', m)}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(required)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_info = pd.concat([pd.DataFrame(required), samples], axis=1)\n",
    "sample_info = pd.DataFrame(sample_info, columns = ['#SampleID', 'Title', 'Anonymized_Name',\n",
    "          'Scientific_Name', 'Taxon_ID', \n",
    "          #'Description', \n",
    "          'Sample_Type', 'Geo_Loc_Name', 'Elevation',\n",
    "          'Env_Biome', 'Env_Feature', 'Env_Material', 'Env_Package', 'Latitude', 'Longitude',\n",
    "          'Collection_Timestamp', 'DNA_Extracted', 'Physical_Specimen_Location', 'Physical_Specimen_Remaining',\n",
    "          'Age', 'Age_Units', 'Host_Subject_ID', 'Host_Taxid', 'Host_Scientific_Name', 'Host_Common_Name',\n",
    "          'Life_Stage', 'Sex', 'Height', 'Height_Units', 'Weight', 'Weight_Units', 'BMI', 'Body_Habitat',\n",
    "          'Body_Site', 'Body_Product', 'GIERACE', 'SITE', 'AMAMPT', 'AMPHIT', 'AMFVT', 'MIDATA', 'V4DATE',\n",
    "          'GIMSTAT', 'MHDIAB', 'MHRHEU1', 'MHOA', 'MHCHF', 'MHMI', 'MHDEPR', 'MHDEPRT', 'TUDRAMT_REVISED', 'PQPSLMED', \n",
    "          'PQPSQUAL', 'PASCORE', 'QLCOMP', 'SLSLPHRS', 'SLEEPHRS', 'PQPEFFCY', 'PQPEFFIC', 'PQPSQI', 'PQBADSLP', 'TURSMOKE', \n",
    "          'M1ADEPR', 'M1BENZO', 'AMAMPT_C1', 'AMFVT_C1', 'AMPHIT_15SD'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(599, 66)"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_info.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#SampleID</th>\n",
       "      <th>Title</th>\n",
       "      <th>Anonymized_Name</th>\n",
       "      <th>Scientific_Name</th>\n",
       "      <th>Taxon_ID</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>Geo_Loc_Name</th>\n",
       "      <th>Elevation</th>\n",
       "      <th>Env_Biome</th>\n",
       "      <th>Env_Feature</th>\n",
       "      <th>...</th>\n",
       "      <th>PQPEFFCY</th>\n",
       "      <th>PQPEFFIC</th>\n",
       "      <th>PQPSQI</th>\n",
       "      <th>PQBADSLP</th>\n",
       "      <th>TURSMOKE</th>\n",
       "      <th>M1ADEPR</th>\n",
       "      <th>M1BENZO</th>\n",
       "      <th>AMAMPT_C1</th>\n",
       "      <th>AMFVT_C1</th>\n",
       "      <th>AMPHIT_15SD</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BI0023</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0023</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>93.33</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>M:Not Applicable</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BI0056</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0056</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>60</td>\n",
       "      <td>3:&lt;65% EFFICIENCY</td>\n",
       "      <td>9.0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BI0131</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0131</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>90</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BI0153</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0153</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>100</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BI0215</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0215</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>88.89</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BI0353</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0353</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>80</td>\n",
       "      <td>1:&gt;=75% TO &lt;=85% EFFICIENCY</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>BI0371</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0371</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>88.89</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0:NO</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BI0372</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0372</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>82.35</td>\n",
       "      <td>1:&gt;=75% TO &lt;=85% EFFICIENCY</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0:NO</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>BI0380</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0380</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>77.78</td>\n",
       "      <td>1:&gt;=75% TO &lt;=85% EFFICIENCY</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>BI0389</td>\n",
       "      <td>MrOS_VitaminD</td>\n",
       "      <td>BI0389</td>\n",
       "      <td>human gut metagenome</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>stool</td>\n",
       "      <td>Birmingham</td>\n",
       "      <td>Not applicable</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>human-associated habitat</td>\n",
       "      <td>...</td>\n",
       "      <td>70</td>\n",
       "      <td>2:&gt;=65% TO &lt;75% EFFICIENCY</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>0:NO</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 66 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  #SampleID          Title Anonymized_Name       Scientific_Name  \\\n",
       "0    BI0023  MrOS_VitaminD          BI0023  human gut metagenome   \n",
       "1    BI0056  MrOS_VitaminD          BI0056  human gut metagenome   \n",
       "2    BI0131  MrOS_VitaminD          BI0131  human gut metagenome   \n",
       "3    BI0153  MrOS_VitaminD          BI0153  human gut metagenome   \n",
       "4    BI0215  MrOS_VitaminD          BI0215  human gut metagenome   \n",
       "5    BI0353  MrOS_VitaminD          BI0353  human gut metagenome   \n",
       "6    BI0371  MrOS_VitaminD          BI0371  human gut metagenome   \n",
       "7    BI0372  MrOS_VitaminD          BI0372  human gut metagenome   \n",
       "8    BI0380  MrOS_VitaminD          BI0380  human gut metagenome   \n",
       "9    BI0389  MrOS_VitaminD          BI0389  human gut metagenome   \n",
       "\n",
       "         Taxon_ID Sample_Type Geo_Loc_Name       Elevation    Env_Biome  \\\n",
       "0  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "1  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "2  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "3  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "4  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "5  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "6  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "7  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "8  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "9  Not applicable       stool   Birmingham  Not applicable  urban biome   \n",
       "\n",
       "                Env_Feature     ...     PQPEFFCY                     PQPEFFIC  \\\n",
       "0  human-associated habitat     ...        93.33            0:>85% EFFICIENCY   \n",
       "1  human-associated habitat     ...           60            3:<65% EFFICIENCY   \n",
       "2  human-associated habitat     ...           90            0:>85% EFFICIENCY   \n",
       "3  human-associated habitat     ...          100            0:>85% EFFICIENCY   \n",
       "4  human-associated habitat     ...        88.89            0:>85% EFFICIENCY   \n",
       "5  human-associated habitat     ...           80  1:>=75% TO <=85% EFFICIENCY   \n",
       "6  human-associated habitat     ...        88.89            0:>85% EFFICIENCY   \n",
       "7  human-associated habitat     ...        82.35  1:>=75% TO <=85% EFFICIENCY   \n",
       "8  human-associated habitat     ...        77.78  1:>=75% TO <=85% EFFICIENCY   \n",
       "9  human-associated habitat     ...           70   2:>=65% TO <75% EFFICIENCY   \n",
       "\n",
       "  PQPSQI PQBADSLP          TURSMOKE M1ADEPR M1BENZO AMAMPT_C1  AMFVT_C1  \\\n",
       "0    4.0    0: No  M:Not Applicable   0: No   0: No       NaN       NaN   \n",
       "1    9.0   1: Yes            1:PAST   0: No   0: No       1.0       1.0   \n",
       "2    2.0    0: No            1:PAST   0: No   0: No       1.0       1.0   \n",
       "3    3.0    0: No            1:PAST   0: No   0: No       1.0       1.0   \n",
       "4    1.0    0: No            1:PAST   0: No   0: No       1.0       0.0   \n",
       "5    6.0   1: Yes            1:PAST   0: No   0: No       1.0       1.0   \n",
       "6    5.0    0: No              0:NO   0: No   0: No       1.0       0.0   \n",
       "7    3.0    0: No              0:NO  1: Yes   0: No       0.0       0.0   \n",
       "8   11.0   1: Yes            1:PAST   0: No   0: No       0.0       0.0   \n",
       "9    6.0   1: Yes              0:NO  1: Yes   0: No       1.0       1.0   \n",
       "\n",
       "  AMPHIT_15SD  \n",
       "0         NaN  \n",
       "1         0.0  \n",
       "2         0.0  \n",
       "3        -1.0  \n",
       "4        -1.0  \n",
       "5         0.0  \n",
       "6         0.0  \n",
       "7        -1.0  \n",
       "8         0.0  \n",
       "9         0.0  \n",
       "\n",
       "[10 rows x 66 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_info.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Match sample_name in two documents"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "id_prep=list(prep_info.loc[:,'#SampleID'].values)\n",
    "id_sample=list(sample_info.loc[:,'#SampleID'].values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "# samples with metadata yet no sequencing data\n",
    "for i in id_sample:\n",
    "    if i not in id_prep:\n",
    "        print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# samples with sequencing data yet no metadata\n",
    "no_sample_info = []\n",
    "for j in id_prep:\n",
    "    if j not in id_sample:\n",
    "        no_sample_info.append(j)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "['PO7100']\n"
     ]
    }
   ],
   "source": [
    "print(len(no_sample_info))\n",
    "print(no_sample_info)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Drop un-needed samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "# exclude PO7100, as it has no microbiome data (see Lily Liu's email on 08/07/2017)\n",
    "prep_info = prep_info.loc[~prep_info['#SampleID'].isin(['PO7100'])]\n",
    "sample_info = sample_info.loc[~sample_info['#SampleID'].isin(['PO7100'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(599, 11)\n",
      "(599, 66)\n"
     ]
    }
   ],
   "source": [
    "print(prep_info.shape)\n",
    "print(sample_info.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## create Qiita prep and Qiime mapping files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#SampleID</th>\n",
       "      <th>Barcode</th>\n",
       "      <th>LinkerPrimerSequence</th>\n",
       "      <th>Description</th>\n",
       "      <th>Experiment_Design_Description</th>\n",
       "      <th>Library_Construction_Protocol</th>\n",
       "      <th>Linker</th>\n",
       "      <th>Platform</th>\n",
       "      <th>Center_Name</th>\n",
       "      <th>Center_Project</th>\n",
       "      <th>Instrument_Model</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BI0023</td>\n",
       "      <td>TCTGGTGACATT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BI0056</td>\n",
       "      <td>CAAGCATGCCTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BI0131</td>\n",
       "      <td>CTATTTGCGACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BI0153</td>\n",
       "      <td>ATCGGCGTTACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0153.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BI0215</td>\n",
       "      <td>CCTCTCGTGATC</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0215.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #SampleID       Barcode  LinkerPrimerSequence       Description  \\\n",
       "0    BI0023  TCTGGTGACATT  GGACTACHVGGGTWTCTAAT  Orwoll.BI0023.BI   \n",
       "1    BI0056  CAAGCATGCCTA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0056.BI   \n",
       "2    BI0131  CTATTTGCGACA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0131.BI   \n",
       "3    BI0153  ATCGGCGTTACA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0153.BI   \n",
       "4    BI0215  CCTCTCGTGATC  GGACTACHVGGGTWTCTAAT  Orwoll.BI0215.BI   \n",
       "\n",
       "                       Experiment_Design_Description  \\\n",
       "0  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "1  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "2  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "3  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "4  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "\n",
       "  Library_Construction_Protocol Linker  Platform Center_Name Center_Project  \\\n",
       "0                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "1                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "2                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "3                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "4                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "\n",
       "  Instrument_Model  \n",
       "0   Illumina MiSeq  \n",
       "1   Illumina MiSeq  \n",
       "2   Illumina MiSeq  \n",
       "3   Illumina MiSeq  \n",
       "4   Illumina MiSeq  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prep_info.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "mapping = pd.merge(prep_info, sample_info, on='#SampleID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(599, 11)\n",
      "(599, 66)\n",
      "(599, 76)\n"
     ]
    }
   ],
   "source": [
    "print(prep_info.shape)\n",
    "print(sample_info.shape)\n",
    "print(mapping.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "#mapping.rename(columns={'Sample_Name': '#SampleID'}, inplace=True)\n",
    "mapping.rename(columns={'Barcode': 'BarcodeSequence'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(599, 76)"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapping.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#SampleID</th>\n",
       "      <th>BarcodeSequence</th>\n",
       "      <th>LinkerPrimerSequence</th>\n",
       "      <th>Description</th>\n",
       "      <th>Experiment_Design_Description</th>\n",
       "      <th>Library_Construction_Protocol</th>\n",
       "      <th>Linker</th>\n",
       "      <th>Platform</th>\n",
       "      <th>Center_Name</th>\n",
       "      <th>Center_Project</th>\n",
       "      <th>...</th>\n",
       "      <th>PQPEFFCY</th>\n",
       "      <th>PQPEFFIC</th>\n",
       "      <th>PQPSQI</th>\n",
       "      <th>PQBADSLP</th>\n",
       "      <th>TURSMOKE</th>\n",
       "      <th>M1ADEPR</th>\n",
       "      <th>M1BENZO</th>\n",
       "      <th>AMAMPT_C1</th>\n",
       "      <th>AMFVT_C1</th>\n",
       "      <th>AMPHIT_15SD</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BI0023</td>\n",
       "      <td>TCTGGTGACATT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>...</td>\n",
       "      <td>93.33</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>M:Not Applicable</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BI0056</td>\n",
       "      <td>CAAGCATGCCTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>...</td>\n",
       "      <td>60</td>\n",
       "      <td>3:&lt;65% EFFICIENCY</td>\n",
       "      <td>9.0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BI0131</td>\n",
       "      <td>CTATTTGCGACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>...</td>\n",
       "      <td>90</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BI0153</td>\n",
       "      <td>ATCGGCGTTACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0153.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>...</td>\n",
       "      <td>100</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BI0215</td>\n",
       "      <td>CCTCTCGTGATC</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>Orwoll.BI0215.BI</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>...</td>\n",
       "      <td>88.89</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 76 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  #SampleID BarcodeSequence  LinkerPrimerSequence       Description  \\\n",
       "0    BI0023    TCTGGTGACATT  GGACTACHVGGGTWTCTAAT  Orwoll.BI0023.BI   \n",
       "1    BI0056    CAAGCATGCCTA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0056.BI   \n",
       "2    BI0131    CTATTTGCGACA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0131.BI   \n",
       "3    BI0153    ATCGGCGTTACA  GGACTACHVGGGTWTCTAAT  Orwoll.BI0153.BI   \n",
       "4    BI0215    CCTCTCGTGATC  GGACTACHVGGGTWTCTAAT  Orwoll.BI0215.BI   \n",
       "\n",
       "                       Experiment_Design_Description  \\\n",
       "0  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "1  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "2  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "3  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "4  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "\n",
       "  Library_Construction_Protocol Linker  Platform Center_Name Center_Project  \\\n",
       "0                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "1                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "2                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "3                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "4                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "\n",
       "      ...     PQPEFFCY           PQPEFFIC PQPSQI PQBADSLP          TURSMOKE  \\\n",
       "0     ...        93.33  0:>85% EFFICIENCY    4.0    0: No  M:Not Applicable   \n",
       "1     ...           60  3:<65% EFFICIENCY    9.0   1: Yes            1:PAST   \n",
       "2     ...           90  0:>85% EFFICIENCY    2.0    0: No            1:PAST   \n",
       "3     ...          100  0:>85% EFFICIENCY    3.0    0: No            1:PAST   \n",
       "4     ...        88.89  0:>85% EFFICIENCY    1.0    0: No            1:PAST   \n",
       "\n",
       "  M1ADEPR M1BENZO AMAMPT_C1 AMFVT_C1 AMPHIT_15SD  \n",
       "0   0: No   0: No       NaN      NaN         NaN  \n",
       "1   0: No   0: No       1.0      1.0         0.0  \n",
       "2   0: No   0: No       1.0      1.0         0.0  \n",
       "3   0: No   0: No       1.0      1.0        -1.0  \n",
       "4   0: No   0: No       1.0      0.0        -1.0  \n",
       "\n",
       "[5 rows x 76 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapping.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "# move 'Description' to the end\n",
    "cols = list(mapping)\n",
    "cols.insert(mapping.shape[1], cols.pop(cols.index('Description')))\n",
    "mapping = mapping.loc[:, cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#SampleID</th>\n",
       "      <th>BarcodeSequence</th>\n",
       "      <th>LinkerPrimerSequence</th>\n",
       "      <th>Experiment_Design_Description</th>\n",
       "      <th>Library_Construction_Protocol</th>\n",
       "      <th>Linker</th>\n",
       "      <th>Platform</th>\n",
       "      <th>Center_Name</th>\n",
       "      <th>Center_Project</th>\n",
       "      <th>Instrument_Model</th>\n",
       "      <th>...</th>\n",
       "      <th>PQPEFFIC</th>\n",
       "      <th>PQPSQI</th>\n",
       "      <th>PQBADSLP</th>\n",
       "      <th>TURSMOKE</th>\n",
       "      <th>M1ADEPR</th>\n",
       "      <th>M1BENZO</th>\n",
       "      <th>AMAMPT_C1</th>\n",
       "      <th>AMFVT_C1</th>\n",
       "      <th>AMPHIT_15SD</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BI0023</td>\n",
       "      <td>TCTGGTGACATT</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>M:Not Applicable</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Orwoll.BI0023.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BI0056</td>\n",
       "      <td>CAAGCATGCCTA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>3:&lt;65% EFFICIENCY</td>\n",
       "      <td>9.0</td>\n",
       "      <td>1: Yes</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Orwoll.BI0056.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BI0131</td>\n",
       "      <td>CTATTTGCGACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Orwoll.BI0131.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BI0153</td>\n",
       "      <td>ATCGGCGTTACA</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>Orwoll.BI0153.BI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BI0215</td>\n",
       "      <td>CCTCTCGTGATC</td>\n",
       "      <td>GGACTACHVGGGTWTCTAAT</td>\n",
       "      <td>16S stool samples sequenced for MrOS Vitamin D...</td>\n",
       "      <td>16S rRNA v4</td>\n",
       "      <td>GT</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>BI</td>\n",
       "      <td>MrOS</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>0:&gt;85% EFFICIENCY</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1:PAST</td>\n",
       "      <td>0: No</td>\n",
       "      <td>0: No</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>Orwoll.BI0215.BI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 76 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  #SampleID BarcodeSequence  LinkerPrimerSequence  \\\n",
       "0    BI0023    TCTGGTGACATT  GGACTACHVGGGTWTCTAAT   \n",
       "1    BI0056    CAAGCATGCCTA  GGACTACHVGGGTWTCTAAT   \n",
       "2    BI0131    CTATTTGCGACA  GGACTACHVGGGTWTCTAAT   \n",
       "3    BI0153    ATCGGCGTTACA  GGACTACHVGGGTWTCTAAT   \n",
       "4    BI0215    CCTCTCGTGATC  GGACTACHVGGGTWTCTAAT   \n",
       "\n",
       "                       Experiment_Design_Description  \\\n",
       "0  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "1  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "2  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "3  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "4  16S stool samples sequenced for MrOS Vitamin D...   \n",
       "\n",
       "  Library_Construction_Protocol Linker  Platform Center_Name Center_Project  \\\n",
       "0                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "1                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "2                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "3                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "4                   16S rRNA v4     GT  Illumina          BI           MrOS   \n",
       "\n",
       "  Instrument_Model        ...                  PQPEFFIC PQPSQI PQBADSLP  \\\n",
       "0   Illumina MiSeq        ...         0:>85% EFFICIENCY    4.0    0: No   \n",
       "1   Illumina MiSeq        ...         3:<65% EFFICIENCY    9.0   1: Yes   \n",
       "2   Illumina MiSeq        ...         0:>85% EFFICIENCY    2.0    0: No   \n",
       "3   Illumina MiSeq        ...         0:>85% EFFICIENCY    3.0    0: No   \n",
       "4   Illumina MiSeq        ...         0:>85% EFFICIENCY    1.0    0: No   \n",
       "\n",
       "           TURSMOKE M1ADEPR M1BENZO AMAMPT_C1 AMFVT_C1 AMPHIT_15SD  \\\n",
       "0  M:Not Applicable   0: No   0: No       NaN      NaN         NaN   \n",
       "1            1:PAST   0: No   0: No       1.0      1.0         0.0   \n",
       "2            1:PAST   0: No   0: No       1.0      1.0         0.0   \n",
       "3            1:PAST   0: No   0: No       1.0      1.0        -1.0   \n",
       "4            1:PAST   0: No   0: No       1.0      0.0        -1.0   \n",
       "\n",
       "        Description  \n",
       "0  Orwoll.BI0023.BI  \n",
       "1  Orwoll.BI0056.BI  \n",
       "2  Orwoll.BI0131.BI  \n",
       "3  Orwoll.BI0153.BI  \n",
       "4  Orwoll.BI0215.BI  \n",
       "\n",
       "[5 rows x 76 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapping.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['#SampleID', 'BarcodeSequence', 'LinkerPrimerSequence',\n",
       "       'Experiment_Design_Description', 'Library_Construction_Protocol',\n",
       "       'Linker', 'Platform', 'Center_Name', 'Center_Project',\n",
       "       'Instrument_Model', 'Title', 'Anonymized_Name', 'Scientific_Name',\n",
       "       'Taxon_ID', 'Sample_Type', 'Geo_Loc_Name', 'Elevation', 'Env_Biome',\n",
       "       'Env_Feature', 'Env_Material', 'Env_Package', 'Latitude', 'Longitude',\n",
       "       'Collection_Timestamp', 'DNA_Extracted', 'Physical_Specimen_Location',\n",
       "       'Physical_Specimen_Remaining', 'Age', 'Age_Units', 'Host_Subject_ID',\n",
       "       'Host_Taxid', 'Host_Scientific_Name', 'Host_Common_Name', 'Life_Stage',\n",
       "       'Sex', 'Height', 'Height_Units', 'Weight', 'Weight_Units', 'BMI',\n",
       "       'Body_Habitat', 'Body_Site', 'Body_Product', 'GIERACE', 'SITE',\n",
       "       'AMAMPT', 'AMPHIT', 'AMFVT', 'MIDATA', 'V4DATE', 'GIMSTAT', 'MHDIAB',\n",
       "       'MHRHEU1', 'MHOA', 'MHCHF', 'MHMI', 'MHDEPR', 'MHDEPRT',\n",
       "       'TUDRAMT_REVISED', 'PQPSLMED', 'PQPSQUAL', 'PASCORE', 'QLCOMP',\n",
       "       'SLSLPHRS', 'SLEEPHRS', 'PQPEFFCY', 'PQPEFFIC', 'PQPSQI', 'PQBADSLP',\n",
       "       'TURSMOKE', 'M1ADEPR', 'M1BENZO', 'AMAMPT_C1', 'AMFVT_C1',\n",
       "       'AMPHIT_15SD', 'Description'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapping.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "# convert to proper variable types (categorical or continuous)\n",
    "vars_cat = np.array(['BarcodeSequence', 'LinkerPrimerSequence', 'Experiment_Design_Description',\n",
    "             'Library_Construction_Protocol', 'Linker', 'Platform', 'Center_Name', 'Center_Project', 'Instrument_Model',\n",
    "             'Title', 'Anonymized_Name', 'Scientific_Name', 'Taxon_ID', 'Sample_Type', 'Geo_Loc_Name', 'Elevation', 'Env_Biome',\n",
    "             'Env_Feature', 'Env_Material', 'Env_Package', 'Collection_Timestamp', 'DNA_Extracted', 'Physical_Specimen_Location',\n",
    "             'Physical_Specimen_Remaining', 'Age_Units', 'Host_Subject_ID', 'Host_Taxid','Host_Scientific_Name', 'Host_Common_Name',\n",
    "             'Life_Stage', 'Sex', 'Height_Units', 'Weight_Units', 'Body_Habitat', 'Body_Site', 'Body_Product', 'GIERACE', 'SITE',\n",
    "             'SLEEPHRS', 'MIDATA', 'AMPHIT_15SD', 'MHDIAB', 'MHRHEU1', 'MHOA', 'MHCHF', 'MHMI', 'MHDEPR', 'GIMSTAT',\n",
    "             'MHDEPRT', 'TUDRAMT_REVISED', 'PQPSQUAL', 'QLCOMP', 'PQBADSLP', 'TURSMOKE', 'M1ADEPR', 'M1BENZO', 'Description',\n",
    "             'AMAMPT_C1', 'AMFVT_C1', 'V4DATE', 'PQPSLMED', 'PQPEFFIC'])\n",
    "vars_cts = np.array(['Latitude', 'Longitude', 'Age', 'Height', 'Weight', 'BMI', 'PASCORE', 'AMAMPT', \n",
    "             'AMPHIT', 'AMFVT',  'PQPSQI', 'PQPEFFCY', 'SLSLPHRS'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = mapping.copy()\n",
    "df[vars_cts] = df[vars_cts].apply(pd.to_numeric, errors='coerce')\n",
    "df[vars_cat] = df[vars_cat].apply(lambda x: x.astype('category'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "prep_info.to_csv('../data/prep_sleep.txt', sep= '\\t', index=False)\n",
    "df.to_csv('../data/mapping_sleep.txt', sep= '\\t', na_rep='Missing:not collected', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
